core/src/main/java/io/grpc/EquivalentAddressGroup.java

```diff
@@ -32,7 +34,7 @@
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1770")
 public final class EquivalentAddressGroup {
 
-  private final List<SocketAddress> addrs;
+  private final List<ProxySocketAddress> addrs;
```

Comment:
No, no, no. This should be something off-to-the-side, not explicit at every layer. We only want two pieces of code to be aware of this: the resolver and the transport. Everything else should be ignorant of it.

It's important that is isn't "baked" into the API like this, because as it is, "there can only be one" such "special" SocketAddress. We want to allow other such addresses as there is need, but we only want those pieces of code that need to care to care.